### DISCUSSION TITLE
BP_VRoidRuntimeAuthActorを使用したキャラクター変更後のレベル移動について

### DISCUSSION THREAD

**Discussion by o-hkmr** [pixiv/vroid-sdk-developers#190] [Q&A]
以下URLの「Game Animation Sampleで使用するには」の設定を行い、VRoidAuthLevelから別のレベル（DefaultLevel）へ移動後に変更キャラクターが反映されていることを確認しています。
https://vroid.notion.site/VRoid-SDK-for-Unreal-Engine-21321b5e31b247e99600f270b3414d4e#ad705af284344403b6a7894eefec1ac7

以下URLを参考に任意のLevelに「BP_VRoidRuntimeAuthActor」を配置し、キャラクターを変更できるようになったのですが、OpenLevelで別のレベル（DefaultLevel）に移動した際に変更キャラクターが反映されません。
https://vroid.notion.site/VRoid-SDK-for-Unreal-Engine-21321b5e31b247e99600f270b3414d4e#fbcc958168e44da397e933e28e0f5575
![level_change](https://github.com/user-attachments/assets/5eb69210-0f7b-4951-92bf-885bc17b691d)

BP_VRoidRuntimeAuthActorを使用したキャラクター変更のあとに、別のレベルに移動してもキャラクターを維持したい場合、どのような対応をすればよいでしょうか？

UE5.5、VRoidSDKは1.3.1を使用しています。
サポートいただけると幸いです。

**Comment by PaperSloth** [comment]
現在手元にPCがなく、手元での確認/ノードを共有できず申し訳ないのですが

BP_VRoidRuntimeAuthActorを使用してVrmAssetListを更新しております。
その際に、VRoidGameInstanceSubsystemにVrmAssetListを保存することで更新が可能になるかと思います。

BP_VRoidAuthActorの中でVRoidGameInstanceSubsystemを検索して、使用方法を参考にしていただけると解決できるかもしれません🙇‍♂️

**Reply by o-hkmr** [reply]
アドバイスありがとうございます！
C++のAVRoidAuthActor::RegisterVrmAssetListを参考に、BP_VRoidRuntimeAuthActorのOnSuccessUpdateVrmに以下のノードを追加したところ、任意のレベルでキャラクター変更後にレベル移動してもキャラクターが維持されるようになりました。
![update_vrm](https://github.com/user-attachments/assets/db21f865-a50c-463e-abec-321dde295bc2)

**Reply by PaperSloth** [reply]
ノードの画像も添付いただきありがとうございます！
まさしくその通りです🙌
次回リリース時にはオプション機能等を追加して簡単に保存が行えるように改善しようと思います！

**Reply by mameoPix** [reply]
本日VRoid SDK for UE 1.4.0がリリースされました。
それに伴い、BP_VRoidRuntimeAuthActorにIs Cache Vrmというフラグを追加したことで、本件の問題に対応させていただきました！
https://vroid.notion.site/VRoid-SDK-for-Unreal-Engine-21321b5e31b247e99600f270b3414d4e#fbcc958168e44da397e933e28e0f5575

**Comment by o-hkmr** [comment]
VRoid SDK for UE 1.4.1においてUnreal Editorが正常に起動し、Is Cache VrmフラグをONにすることでLevel遷移後も変更したキャラクターを利用できることを確認しました。
対応ありがとうございました！
